com.unity.shadergraph/Editor/ShaderGraphProjectSettings.cs

```diff
@@ -0,0 +1,91 @@
+using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ShaderGraph
+{
+    [FilePath("ProjectSettings/ShaderGraphProjectSettings.asset", FilePathAttribute.Location.ProjectFolder)]
+    internal class ShaderGraphProjectSettings : ScriptableSingleton<ShaderGraphProjectSettings>
+    {
+        [SerializeField]
+        internal int customInterpolatorErrorThreshold = 32;
+        [SerializeField]
+        internal int customInterpolatorWarningThreshold = 16;
+        internal SerializedObject GetSerializedObject() { return new SerializedObject(this); }
+        internal void Save() { Save(true); }
+        private void OnDisable() { Save(); }
+    }
+
+    class ShaderGraphProjectSettingsProvider : SettingsProvider
+    {
+        private static string kCustomInterpolatorHelpBox = "These options are used to help ShaderGraph users maintain known compatibilities with target platform(s) when using Custom Interpolators.";
+        private static string kCustomInterpolatorDocumentationURL = "https://docs.unity3d.com/Packages/com.unity.shadergraph@latest/index.html?preview=1&subfolder=/manual/";
+
+        private class Styles
+        {
+            public static readonly GUIContent CustomInterpLabel = L10n.TextContent("Custom Interpolator Channel Settings", "");
+            public static readonly GUIContent CustomInterpWarnThresholdLabel = L10n.TextContent("Warning Threshold", "ShaderGraph will warn when this channel limitation is exceeded by a custom interpolator. Range must be between 8 and Error Threshold.");
+            public static readonly GUIContent CustomInterpErrorThresholdLabel = L10n.TextContent("Error Threshold", "ShaderGraph will error when this channel limitation is exceeded by a custom interpolator. Range must be between 8 and 32.");
+            public static readonly GUIContent ReadMore = L10n.TextContent("Read more");
+        }
```

Comment:
@ValGrimm-U3D Need to get an idea what the URL is for Custom Interpolator documentation for thresholds, and whether the above strings are okay or not.

Reply:
I don't yet have a URL as we are still working on the doc (waiting on this very bit settings info, in fact--now I can add that to the doc and get it to editorial : D ), and I'm not yet sure how to create persistent, version-neutral links to use outside of the Shader Graph or core manual doc. I've posted a question to docs training about that.

In the meantime, here's revisions for the two messages (guessing on the "Range" bit, please correct me if I've guessed wrong):

- ShaderGraph displays a warning when the user creates more custom interpolators than permitted by this setting.   The number of interpolators that trigger this warning must be between 8 and Error Threshold.

- ShaderGraph displays an error message when the user tries to create more custom interpolators than permitted by this setting.  The number of interpolators that trigger this error must be between 8 and 32.


(where's the warning and error messages? I couldn't find these to check them)

Reply:
Oh, I know how to do the version neutral, nbd. I just need to know what the name of the docs file is going to be so I can jump the user to that specific spot.

Reply:
Oh, also-- :(, this whole thing has a visual element to it as well. Not sure how you want to get those.

Reply:
Just saw your last comment, not sure what you mean about a visual element?